src/rpc/server/basic.zig

```diff
@@ -182,10 +292,266 @@ pub fn acceptAndServeConnection(server_ctx: *server.Context) !void {
+            }) catch |e| switch (e) {
+                error.ConnectionResetByPeer => return,
+                else => return error.SystemIoError,
+            };
+            return;
+        },
+        else => {
+            writeJsonResponse(request, .{}, .{
+                .jsonrpc = "2.0",
+                .id = null,
+                .@"error" = rpc.response.Error{
+                    .code = .parse_error,
+                    .message = "Invalid json",
+                },
+            }) catch |e| switch (e) {
+                error.ConnectionResetByPeer => return,
+                else => return error.SystemIoError,
+            };
+            return;
+        },
+    };
+
+    const rpc_request = json: {
+        var diag = rpc.Request.Dynamic.ParseDiagnostic.INIT;
+        const result = rpc_request_dyn.parse(json_arena, .{}, &diag) catch |err| {
+            const code: rpc.response.ErrorCode, //
+            const message: []const u8 //
+            = switch (err) {
+                error.OutOfMemory,
+                => |e| return e,
+
+                error.MissingId,
+                error.MissingJsonRpcVersion,
+                error.MissingMethod,
+                error.MissingParams,
+                error.InvalidJsonRpcVersion,
+                => .{ .invalid_request, "Invalid request" },
+
+                error.InvalidMethod,
+                => .{ .method_not_found, "Method not found" },
+
+                error.InvalidParams,
+                error.ParamsLengthMismatch,
+                => .{ .invalid_params, "Invalid method parameters" },
+            };
+            const err_obj: rpc.response.Error = .{
+                .code = code,
+                .message = message,
+            };
+            writeJsonResponse(request, .{}, .{
+                .jsonrpc = "2.0",
+                .id = diag.err.id orelse .null,
+                .@"error" = err_obj,
+            }) catch |e| switch (e) {
+                error.ConnectionResetByPeer => return,
+                else => return error.SystemIoError,
+            };
+            return;
+        };
+        break :json result;
+    };
+
+    switch (rpc_request.method) {
+        .getAccountInfo => |params| {
+            const config: rpc.methods.GetAccountInfo.Config = params.config orelse .{};
+            const encoding = config.encoding orelse .base64;
+            if (config.commitment) |commitment| {
+                std.debug.panic("TODO: handle commitment={s}", .{@tagName(commitment)});
+            }
+
+            const account: sig.accounts_db.AccountsDB.AccountInCacheOrFile, //
+            const account_slot: sig.core.Slot, //
+            var account_lg: sig.accounts_db.AccountsDB.AccountInCacheOrFileLock //
+            = (server_ctx.accountsdb.getSlotAndAccountInSlotRangeWithReadLock(
+                &params.pubkey,
+                // if it's null, it's null, there's no floor to the query.
+                config.minContextSlot orelse null,
```

Comment:
redundant? is this just for documentation? isn't it just a passthrough of the same semantics from one api to the next? doesn't seem super relevant in this context.

Reply:
It seems like an important detail to me to understand the type of this parameter before being passed.

Reply:
Personally, I would just make sure the two APIs are well documented (i.e. in methods.zig and in accountsdb), and not worry about explaining the APIs where they are used (e.g. this file). Seems unnecessary, but I'm fine keeping this comment if you think it's valuable.